src/shared/components/common/icon.tsx

```diff
@@ -16,23 +16,51 @@ export class Icon extends Component<IconProps, any> {
   }
 
   render() {
-    return (
-      <svg
-        className={classNames("icon", this.props.classes, {
-          "icon-inline": this.props.inline,
-          small: this.props.small,
-        })}
-      >
-        <use
-          xlinkHref={`${getStaticDir()}/assets/symbols.svg#icon-${
-            this.props.icon
-          }`}
-        ></use>
-        <div className="visually-hidden">
-          <title>{this.props.icon}</title>
-        </div>
-      </svg>
-    );
+    let iconAltText: string;
+    if (
+      this.props.icon === "plus-square" ||
+      this.props.icon === "minus-square"
+    ) {
+      iconAltText = `${I18NextService.i18n.t("show_content")}`;
+    } else {
+      iconAltText = "";
+    }
+
+    if (iconAltText !== "") {
+      return (
+        <svg
+          className={classNames("icon", this.props.classes, {
+            "icon-inline": this.props.inline,
+            small: this.props.small,
+          })}
+          role="img"
+          aria-describedby={`${this.props.icon}-alt`}
+        >
+          <title id={`${this.props.icon}-alt`}>{iconAltText}</title>
+          <use
+            xlinkHref={`${getStaticDir()}/assets/symbols.svg#icon-${
+              this.props.icon
+            }`}
+          ></use>
+        </svg>
+      );
+    } // use aria-describedby or not
+    else {
+      return (
+        <svg
+          className={classNames("icon", this.props.classes, {
+            "icon-inline": this.props.inline,
+            small: this.props.small,
+          })}
+        >
+          <use
+            xlinkHref={`${getStaticDir()}/assets/symbols.svg#icon-${
+              this.props.icon
+            }`}
+          ></use>
+        </svg>
+      );
+    }
```

Comment:
This can be shortened somewhat. I think the following should work:

```suggestion
    let iconAltText: string;
    if (
      this.props.icon === "plus-square" ||
      this.props.icon === "minus-square"
    ) {
      iconAltText = `${I18NextService.i18n.t("show_content")}`;
    }
    return (
      <svg
        className={classNames("icon", this.props.classes, {
          "icon-inline": this.props.inline,
          small: this.props.small,
        })}
        {...(iconAltText ? {role: "img", "aria-describedby": `${this.props.icon}-alt`} : {})}
      >
        {iconAltText && (<title id={`${this.props.icon}-alt`}>{iconAltText}</title>)}
        <use
          xlinkHref={`${getStaticDir()}/assets/symbols.svg#icon-${
            this.props.icon
          }`}
        ></use>
      </svg>
    );
```

The suggestion code will need to be formatted with prettier, and there may be ways to clean it up further.

Reply:
I accepted your suggestion, thank you. It seems to work on my side.

As someone mostly unfamiliar with Typescript I actually find this code harder to understand. I looked up short hand if statements but it didn't include code blocks, can you tell me what this method is called so I can read more about it? What I'm having trouble following is the meaning of `{...`,  `{}`, and `iconAltText && (code block)}` part. If iconAltText is undefined does the && short circuit and then delete itself?

Reply:
Actually the code failed once uploaded to GitHub because the linter does not like that iconAltText isn't defined. That was one of the reasons I had added the blank `iconAltText=""` before.

Reply:
I put it back and it seems to work. Should the `{...{iconAltText ?` part need to evaluate if iconAltText isn't blank now too?